utils/unittest/mocks/protocol_state.go

```diff
@@ -0,0 +1,187 @@
+package mocks
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/stretchr/testify/mock"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol"
+	protocolmock "github.com/onflow/flow-go/state/protocol/mock"
+	"github.com/onflow/flow-go/storage"
+)
+
+// PS is a mocked version of protocol state, which
```

Comment:
A little bit hesitant on having a mocked Protocol State, but it does make the unit tests much more isolated, which is nice.

Reply:
If we don't implement this mock, we will be using another mock, which more or less have to re-implement a partial behaviors, which can't be reused. 

But now, this mock can be reused and extended. 

Testing engines' unittests that depend on protocol state could just use this mocked version